### PR TITLE
podman search --tls-verify option

### PR DESCRIPTION
**Pull request by haircommander**
added option to patch #914 

Podman search doesn't respect registry.insecure , but I wasn't sure how to test that, so I chose the flag option instead. Test included.

### CODE REVIEW

File: cmd/podman/search.go [hunk 1]
```diff
@@ -101,10 +106,11 @@ func searchCmd(c *cli.Context) error {
...
 	opts := searchOpts{
-		format:  format,
-		noTrunc: c.Bool("no-trunc"),
-		limit:   c.Int("limit"),
-		filter:  c.StringSlice("filter"),
+		format:     format,
+		noTrunc:    c.Bool("no-trunc"),
+		limit:      c.Int("limit"),
+		filter:     c.StringSlice("filter"),
+		skipVerify: !c.BoolT("tls-verify"),
```
Comment (TomSweeneyRedHat) [inline cmd/podman/search.go]: I think in most of our code we call a flag like this tlsVerify and then do the inverse of it's value when we set DockerInsecureSkipTLSVerify.

File: cmd/podman/search.go [hunk 2]
```diff
@@ -155,6 +161,13 @@ func (s *searchParams) headerMap() map[string]string {
 
 func getSearchOutput(term string, registries []string, opts searchOpts, filter searchFilterParams) ([]searchParams, error) {
 	sc := common.GetSystemContext("", "", false)
+
+	// If user flagged to skip verify for HTTP connections, set System Context as such
+	// docker client will retry without TLS after making an attempt with TLS, so it will
+	// not effect normal HTTPS requests
+	if opts.skipVerify {
```
Comment (TomSweeneyRedHat) [inline cmd/podman/search.go]: Suggest changing this to tlsVerify and then the true in the next line to false.

**Comment by TomSweeneyRedHat** [comment]
@haircommander looks like the git title is too long.  Youl'll need to 'git commit --amend -s', change the title in the editor that pops up, then rebase/push.

**Comment by mheon** [comment]
Is adding `tls-verify` enough to resolve this bug? Should we not be respecting insecure registries without explicitly setting a flag?

I think the behavior we settled on for pull/push was: if `tls-verify` is not set, we default to on, unless the registry is insecure, in which case we allow insecure. If `tls-verify` is set to true, we force secure, even for insecure registries. If `tls-verify` is set to false, we allow insecure always. It's basically another bool (I think we called it ForceSecure) that we set true if `tls-verify` is true and `tls-verify` was explicitly set.

**Comment by rhatdan** [comment]
Correct, the original issue was that search was not  handling the insecure registries field in registries.conf, I believe.  The tls-verify flag is just to override the default settings set in registries.conf.

**Comment by haircommander** [comment]
So am I correct in understanding this:

if the --registry flag is passed, podman search only searches that registry. if that registry string matches one in the insecure registries, allow for http. 

Also: without a --registry flag, should there be a more fine grained checking of insecure vs secure? I.e. If one of my default registries is an insecure one, but the rest aren't, should I skip TLS, or only do so if the user specifically consents to checking the insecure default registries?

the issue posted #914 only talks about this search with --registry, but I want to make sure I don't mess up behavior for the whole command by changing what is necessary to allow for --registry to properly search as an insecure registry.

**Comment by haircommander** [comment]
please ignore me drowining in git hell. I will fix this on monday...

**Pull request closed without merging**